matching-notification-rest/src/main/java/org/phenotips/matchingnotification/rest/internal/DefaultMatchingNotificationResource.java

```diff
@@ -405,4 +405,31 @@ private boolean isCurrentUserAdmin()
         return this.auth.hasAccess(this.users.getCurrentUser(), Right.ADMIN,
             this.resolver.resolve("", EntityType.WIKI));
     }
+
+    @Override
+    public Response getMatch(Long matchId)
+    {
+        if (matchId == null) {
+            this.slf4Jlogger.error("The requested match id is blank");
+            return Response.status(Response.Status.BAD_REQUEST).build();
+        }
+
+        try {
+            PatientMatch match = this.matchingNotificationManager.getMatch(matchId);
+            JSONObject matchJson = new JSONObject();
+            matchJson.put("results", match != null ? match.toJSON() : null);
```

Comment:
i wouldn't overcomplicate this and would just return `match.toJSON()` as the response JSON, with no wrappers and results fields, similar to how patient REST GET request works